pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -161,6 +163,10 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		var ok bool
 		newRc, ok = obj.(*api.ReplicationController)
 		if !ok {
+			if _, kind, err := typer.ObjectVersionAndKind(obj); err == nil {
+				return cmdutil.UsageError(cmd, "%s contains a %s not a ReplicationController", filename, kind)
+			}
+			glog.V(4).Infof("Object %#v is not a ReplicationController", obj)
```

Comment:
Right, if we can figure out what Type the object is, we don't really need that verbose printer, it's the error handler in case we can't actually figure out what the type is, at least we get something.